### DISCUSSION TITLE
Do I need to unsubscribe from an observable that completed or errored?

### DISCUSSION THREAD

**Discussion by crumbyshirt** [ReactiveX/rxjs#7195] [Q&A] [answered]
I understand that it is good practice to unsubscribe from an observable whenever the subscription is no longer needed. What I want to know is will my subscription implicitly unsubscribe from the observable it was subscribed to when that observable completes?

**Comment by demensky** [comment, marked as answer]
No need.

```ts
const o = new Observable((subscriber) => {
  setTimeout(() => {
    subscriber.complete(); //  log 'complete' & log 'teardown'
  }, 1000);

  return () => {
    console.log('teardown');
  };
});

const s = o.subscribe({
  complete: () => {
    console.log('complete');
  },
});

setTimeout(() => {
  s.unsubscribe(); // log nothing
}, 2000);
```

**Reply by crumbyshirt** [reply]
Thanks, I put that in a stackblitz to see it in action - https://stackblitz.com/edit/rxjs-jmkbbw?devtoolsheight=60&file=index.ts
